packages/devtools_app/lib/src/shared/ui/filter.dart

```diff
@@ -515,29 +508,34 @@ class _StandaloneFilterFieldState<T> extends State<StandaloneFilterField<T>>
     return Row(
       children: [
         Expanded(
-          child: DevToolsClearableTextField(
-            autofocus: true,
-            labelText: 'Filter',
-            controller: queryTextFieldController,
-            additionalSuffixActions: [
-              DevToolsToggleButton(
-                icon: Codicons.regex,
-                message: 'Use regular expressions',
-                outlined: false,
-                isSelected: useRegExp,
-                onPressed: () => setState(() {
-                  widget.controller.useRegExp.value = !useRegExp;
+          child: ValueListenableBuilder<bool>(
+            valueListenable: widget.controller.useRegExp,
+            builder: (context, useRegExp, _) {
+              return DevToolsClearableTextField(
+                autofocus: true,
+                labelText: 'Filter',
+                controller: queryTextFieldController,
+                additionalSuffixActions: [
+                  DevToolsToggleButton(
+                    icon: Codicons.regex,
+                    message: 'Use regular expressions',
+                    outlined: false,
+                    isSelected: useRegExp,
+                    onPressed: () => setState(() {
+                      widget.controller.useRegExp.value = !useRegExp;
```

Comment:
Won't this trigger two rebuilds? One when you update the `ValueListenable` and one when `setState` is called. Maybe the `setState` call can be removed?